lms/djangoapps/instructor/enrollment.py

```diff
@@ -286,36 +291,53 @@ def reset_student_attempts(course_id, student, module_state_key, requesting_user
             text_type(module_state_key),
         )
 
-    module_to_reset = StudentModule.objects.get(
-        student_id=student.id,
-        course_id=course_id,
-        module_state_key=module_state_key
-    )
-
-    if delete_module:
-        module_to_reset.delete()
-        create_new_event_transaction_id()
-        set_event_transaction_type(grades_events.STATE_DELETED_EVENT_TYPE)
-        tracker.emit(
-            six.text_type(grades_events.STATE_DELETED_EVENT_TYPE),
-            {
-                'user_id': six.text_type(student.id),
-                'course_id': six.text_type(course_id),
-                'problem_id': six.text_type(module_state_key),
-                'instructor_id': six.text_type(requesting_user.id),
-                'event_transaction_id': six.text_type(get_event_transaction_id()),
-                'event_transaction_type': six.text_type(grades_events.STATE_DELETED_EVENT_TYPE),
-            }
-        )
-        if not submission_cleared:
-            _fire_score_changed_for_block(
-                course_id,
-                student,
-                block,
-                module_state_key,
+    def _reset_or_delete_module(studentmodule):
+        if delete_module:
+            studentmodule.delete()
+            create_new_event_transaction_id()
+            set_event_transaction_type(grades_events.STATE_DELETED_EVENT_TYPE)
+            tracker.emit(
+                six.text_type(grades_events.STATE_DELETED_EVENT_TYPE),
+                {
+                    'user_id': six.text_type(student.id),
+                    'course_id': six.text_type(course_id),
+                    'problem_id': six.text_type(module_state_key),
+                    'instructor_id': six.text_type(requesting_user.id),
+                    'event_transaction_id': six.text_type(get_event_transaction_id()),
+                    'event_transaction_type': six.text_type(grades_events.STATE_DELETED_EVENT_TYPE),
+                }
             )
+            if not submission_cleared:
+                _fire_score_changed_for_block(
+                    course_id,
+                    student,
+                    block,
+                    module_state_key,
+                )
+        else:
+            _reset_module_attempts(studentmodule)
+
+    team = None
+    if teams_enabled:
+        from lms.djangoapps.teams.api import get_team_for_user_course_topic
+        team = get_team_for_user_course_topic(student, str(course_id), selected_teamset_id)
+    if team:
+        modules_to_reset = StudentModule.objects.filter(
+            student__teams=team,
+            course_id=course_id,
+            module_state_key=module_state_key
+        )
+        for module_to_reset in modules_to_reset:
+            _reset_or_delete_module(module_to_reset)
```

Comment:
What is the limit on team size? Could there be a team with 100K students in it?

Reply:
Good point. There is no "limit" on team size, if a teacher for whatever reason specifies a huge number, we allow it. The default size for an open team (that a user can join and leave at their own discretion) is 50

Reply:
Could we add a cap? If there's no cap at all, then someone, somewhere is going to make a team with 100K students in it, press reload a bunch of times on this task, and bring down a few web workers.

Reply:
FWIW, the largest I'm seeing right now is 368 in one team.